jspwiki-cache/src/main/java/org/apache/wiki/cache/CachingManager.java

```diff
@@ -121,4 +122,12 @@ public interface CachingManager {
      */
     void remove( String cacheName, Serializable key );
 
+    /**
+     * Register a listener to a cache
+     *
+     * @param cacheName The cache to which the listener should be registered
+     * @param listener cache listener to register
+     * @return true if the listener is being added and was not already added
+     */
+    boolean registerListener( String cacheName, CacheEventListener listener );
```

Comment:
Ah, your are right, that CacheEventListener is to specific for the CachingManager. But I think a method like updateAllRequestedFlag(AtomicBoolean flag) is also a bit to specific, no? It is a random implementation/optimization detail of the Caching[Attachment]Provider the CachingManager Interface shouldn't have to know about.

How about we add a generic CacheEvent for jspwiki, that is probably a bit simpler than the one from EhCache and when registering to the specific CachingManager, like the EhcacheCachingManager, we can wrap and delegate to that JSPWikiCacheEvent with the Ehcache (or other) specific implementation of the CacheEvent. 
Regarding the method attributes in CacheEvent, instead of the Ehcache we could use 'String cacheName' and instead of Element we could use 'Serializable name' and 'T or Object object'

If you are okay with this, I'm happy to adapt the pull request accordingly.

Reply:
Hi!

I thought of throwing an event too, but given that one class is a direct dependency of the other, it seemed too much indirection to me. I agree that updateAllRequestedFlag is also too specific. May be something like `registerListener( String cacheName, String listener, T... args)`? `listener` could be used to decide which kind of listener build for a given `cacheName`, with all the required data passed through `args`. WDYT? 

Either going the events way or going through the registerListener method would fine to me, my main concern is that EhCache classes should not appear outside `EhcacheCachingManager`

cheers,
juan pablo